wannierberri/__utility.py

```diff
@@ -285,34 +289,65 @@ def transform(self,AAA_K):
             raise RuntimeError("FFT.transform should not be called for slow FT")
         else :
             raise ValueError("Unknown type of Fourier transform :''".format(self.lib)) 
-
-    def __call__(self,AAA_R,hermitian=False,antihermitian=False,reshapeKline=True):
+    
+    @Lazy
+    def exponent(self):
+        '''
+        exponent for Fourier transform exp(1j*k*R)
+        '''
+        return [np.exp(2j*np.pi/self.NKFFT[i])**np.arange(self.NKFFT[i]) for i in range(3)]
+    @Lazy
+    def exponent_wc(self): 
+        '''
+        additional exponent for Fourier transform under convention 1, exp(1j*k(tau_j - tau_i))
+        '''
+        w_centres_diff = np.array([[j-i for j in self.wannier_centres_reduced] for i in self.wannier_centres_reduced])
+        def exp_par(ii,jj,i):#k1 k2 k3 partial of exponent_wc
+            return np.prod(np.exp(2j*np.pi*w_centres_diff[ii,jj,i]/self.NKFFT[i])**np.arange(self.NKFFT[i]))
```

Comment:
Why product??  exponent_wc should be different for the different k-points of the grid. Instead, you multiply them all, and use the same for all other k-points. 

We call it "temperature averaged over the hospital"